lib/publiccloud/azure.pm

```diff
@@ -72,14 +73,24 @@ sub find_img {
     my $container = $self->container;
     my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT', 'eisleqaopenqa');
     my $key = $self->get_storage_account_keys($storage_account);
-    my $cmd_show = "az storage blob show --account-key $key -o json " .
-      "--container-name '$container' --account-name '$storage_account' --name '$name' " .
-      '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"';
-    record_info('BLOB INFO', script_output($cmd_show, proceed_on_failure => 1));
 
-    my $json = script_output("az image show --resource-group " . $self->resource_group . " --name $name", 60, proceed_on_failure => 1);
+    $json = script_output("az storage blob show --account-key $key -o json " .
+          "--container-name '$container' --account-name '$storage_account' --name '$name' " .
+          '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"',
+        proceed_on_failure => 1);
+    record_info('BLOB INFO', $json);
+    eval {
+        $md5 = decode_azure_json($json)->{md5};
+    };
+    record_info('BLOB NOT-FOUND', "Cannot find blob $name. Need to upload it.\n$@") if ($@);
```

Comment:
AFAICS this information is shown, also if the blob would exist, no?

Reply:
If blob exists there is no error in `eval` and therefore the `$@` is not present.

Reply:
Oh, there is this tiny little `if ($@);` at the end.

I'm adding this to the list of reasons why this is syntactic sin perls commits on a daily basis :facepalm: